### PR TITLE
[WIP]: e-values KO examples and utils test

### PR DESCRIPTION
**Pull request by achamma723**
This PR restores the PR related to KO examples and unit tests.

### CODE REVIEW

**Comment by achamma723** [comment]
@bthirion, I think this PR from the old repo is ready to merge (What Alexandre was working on). What do you think?

File: hidimstat/test/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    null_p_values = np.linspace(1.0e-6, 1 - 1.0e-6, 100)
+    null_e_values = 1 / null_p_values
+
+    null_bhq_cutoff = fdr_threshold(null_p_values, fdr=0.1, method="bhq")
+    null_bhy_cutoff = fdr_threshold(null_p_values, fdr=0.1, method="bhy")
+    null_ebh_cutoff = fdr_threshold(null_e_values, fdr=0.1, method="ebh")
+
+    # Test BHq on null data
+    assert len(null_p_values[null_p_values <= null_bhq_cutoff]) <= 1
```
Comment (bthirion) [inline hidimstat/test/test_utils.py:40]: I'm not sure what we're testing here

File: hidimstat/test/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    # Test BHq on null data
+    assert len(null_p_values[null_p_values <= null_bhy_cutoff]) <= 1
+
+    # Test eBH on null data
+    assert len(null_e_values[null_e_values >= null_ebh_cutoff]) <= 1
+
+
+def test_cal_fdp_power():
+    p_values = np.linspace(1.0e-6, 1 - 1.0e-6, 100)
```
Comment (bthirion) [inline hidimstat/test/test_utils.py:50]: one line docstring ?

File: hidimstat/test/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    non_zero_index = np.concatenate([np.arange(18), [35, 36, 37]])
+
+    fdp, power = cal_fdp_power(selected, non_zero_index)
+
+    assert fdp == 2 / len(selected)
+    assert power == 18 / len(non_zero_index)
+
+
+def test_quantile_aggregation():
+    col = np.arange(11)
```
Comment (bthirion) [inline hidimstat/test/test_utils.py:64]: same here

File: hidimstat/utils.py [hunk 4]
```diff
@@ -9,6 +9,9 @@
 
 
 def quantile_aggregation(pvals, gamma=0.5, gamma_min=0.05, adaptive=False):
+    # if pvalues are one-dimensional, do nothing
+    if pvals.shape[0] == 1:
+        return pvals[0]
```
Comment (bthirion) [inline hidimstat/utils.py:14]: would be good to cover wit a test...

**Review by bthirion — COMMENTED** [review]
Thx ! 
I have a bunch of rather minor comments, but should be mergeable soon.

**Comment by achamma723** [comment]
@bthirion this PR is ready to review

File: hidimstat/utils.py [hunk 5]
```diff
@@ -9,6 +9,9 @@
 
 
 def quantile_aggregation(pvals, gamma=0.5, gamma_min=0.05, adaptive=False):
+    # if pvalues are one-dimensional, do nothing
```
Comment (bthirion) [inline hidimstat/utils.py:12]: A one-line docstring is needed here

**Review by bthirion — COMMENTED** [review]
Almost there !
The example would still be improved, but it's OK to merge it as is.

**Comment by achamma723** [comment]
@bthirion, I think it is ready to be merged

**Comment by bthirion** [comment]
Merging now. Thx !

**Comment by achamma723** [comment]
Hello @bthirion, I think this PR is ready to be merged.

**Pull request merged**